File: LeetCode Solutions/January 2022 LeetCoding Challenge/Pairs of Songs With Total Durations Divisible by 60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        m = [0] * 60
        result = 0
        for t in time:
            result += m[(60 - t % 60) % 60]
            m[t % 60] += 1
        return result

File: LeetCode Solutions/January 2022 LeetCoding Challenge/Pairs of Songs With Total Durations Divisible by 60/Pairs of Songs With Total Durations Divisible by 60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        vector<int> m(60, 0);
        int result = 0;
        for (auto t: time){
            result += m[(60 - t % 60) % 60];
            m[t % 60] += 1;
        }
        return result;
    }
};